scripts/minisat-2.2.1-patch

```diff
@@ -227,3 +227,33 @@ index 9cbbc51..27b9700 100644
  #include <fpu_control.h>
  #endif
  
+diff --git a/minisat/mtl/Vec.h b/minisat/mtl/Vec.h
+--- a/minisat/mtl/Vec.h
++++ b/minisat/mtl/Vec.h
+@@ -96,8 +96,10 @@
+ void vec<T>::capacity(int min_cap) {
+     if (cap >= min_cap) return;
+     int add = imax((min_cap - cap + 1) & ~1, ((cap >> 1) + 2) & ~1);   // NOTE: grow by approximately 3/2
+-    if (add > INT_MAX - cap || (((data = (T*)::realloc(data, (cap += add) * sizeof(T))) == NULL) && errno == ENOMEM))
+-        throw OutOfMemoryException();
++    if (add > INT_MAX - cap)
++      throw OutOfMemoryException();
++
++    data = (T*)xrealloc(data, (cap += add) * sizeof(T));
+  }
+ 
+ 
+diff --git a/minisat/mtl/XAlloc.h b/minisat/mtl/XAlloc.h
+--- a/minisat/mtl/XAlloc.h
++++ b/minisat/mtl/XAlloc.h
+@@ -32,8 +32,9 @@
+ class OutOfMemoryException{};
+ static inline void* xrealloc(void *ptr, size_t size)
+ {
++    errno = 0;
+     void* mem = realloc(ptr, size);
+-    if (mem == NULL && errno == ENOMEM){
++    if (mem == NULL || errno == ENOMEM){
```

Comment:
Just drop the `errno` test completely -- it may have any value when a call succeeds, not necessarily `0`.

Reply:
Done.